Dockerfile

```
@@ -1,38 +1,117 @@
-LABEL org.opencontainers.image.authors="mark.feldhousen@cisa.dhs.gov"
+LABEL org.opencontainers.image.authors="jeremy.frasier@cisa.dhs.gov"
 LABEL org.opencontainers.image.vendor="Cybersecurity and Infrastructure Security Agency"
 
-ARG CISA_UID=421
-ENV CISA_HOME="/home/cisa"
-ENV ECHO_MESSAGE="Hello World from Dockerfile"
+###
+# Install everything we need
+###
+
+###
+# Dependencies
+#
+# Install dependencies are only needed for software installation and
+# will not be included in the final Docker image.
+###
+ENV DEPS \
+    libpq-dev
+ENV INSTALL_DEPS \
+    wget
+RUN apt-get update && \
+    apt-get install --no-install-recommends --no-install-suggests --yes \
+    ${DEPS} ${INSTALL_DEPS}
 
-RUN addgroup --system --gid ${CISA_UID} cisa \
-  && adduser --system --uid ${CISA_UID} --ingroup cisa cisa
+###
+# Setup the unprivileged user and its home directory
+###
+ARG CISA_GID=421
+ARG CISA_UID=${CISA_GID}
+ENV CISA_USER="cisa"
+ENV CISA_GROUP=${CISA_USER}
+ENV CISA_HOME="/home/cisa"
 
-RUN apk --update --no-cache add \
-ca-certificates \
-openssl \
-py-pip
+###
+# Create unprivileged user
+###
+RUN groupadd --system --gid ${CISA_GID} ${CISA_GROUP} && \
+    useradd --system --create-home \
+    --uid ${CISA_UID} --gid ${CISA_GROUP} \
+    --comment "${CISA_USER} user" ${CISA_USER}
 
+# Perform remaining steps as the unprivileged user, from the
+# unprivileged user's home directory
+USER ${CISA_USER}:${CISA_GROUP}
 WORKDIR ${CISA_HOME}
 
-RUN wget -O sourcecode.tgz https://github.com/cisagov/skeleton-python-library/archive/v${VERSION}.tar.gz && \
-  tar xzf sourcecode.tgz --strip-components=1 && \
-  pip install --requirement requirements.txt && \
-  ln -snf /run/secrets/quote.txt src/example/data/secret.txt && \
-  rm sourcecode.tgz
+# Manually set up the virtual environment
+ENV PY_VENV=${CISA_HOME}/.venv
+RUN python -m venv ${PY_VENV}
+ENV PATH="${PY_VENV}/bin:$PATH"
+# Install/upgrade core Python dependencies
+RUN python -m pip install --upgrade pip setuptools wheel
+
+# Download and install guacscanner
+RUN wget -O sourcecode.tgz \
+    https://github.com/cisagov/guacscanner/archive/v${VERSION}.tar.gz && \
+    tar xzf sourcecode.tgz --strip-components=1 && \
+    python -m pip install --requirement requirements.txt && \
+    rm sourcecode.tgz
+
+FROM python:${PY_VERSION}-slim-bullseye AS build-stage
+
+# For a list of pre-defined annotation keys and value types see:
+# https://github.com/opencontainers/image-spec/blob/master/annotations.md
+# Note: Additional labels are added by the build workflow.
+LABEL org.opencontainers.image.authors="jeremy.frasier@cisa.dhs.gov"
+LABEL org.opencontainers.image.vendor="Cybersecurity and Infrastructure Security Agency"
+
+###
+# Dependencies
+###
+ENV DEPS \
+    libpq-dev
+RUN apt-get update && \
+    apt-get install --no-install-recommends --no-install-suggests --yes \
+    ${DEPS}
+
+# Clean up aptitude cruft
+RUN apt-get clean
+RUN rm -rf /var/lib/apt/lists/*
```

```suggestion
RUN apt-get update && \
    apt-get install --no-install-recommends --no-install-suggests --yes \
    ${DEPS} && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*
```

Consider unifying these two `RUN` instructions.  This will reduce the number of layers in the image, and reduce its final size.  As it stands, the cruft will still be in there, but just shadowed by the UnionFS. 

https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers

---

Please enjoy commit 82edf1c!

---

Stellar! 🌟